### DISCUSSION TITLE
ESP32-3248S035C Board Support

### DISCUSSION THREAD

**Discussion by RossiRPI** [HASwitchPlate/openHASP#384] [Hardware]
Any chance we can get the GT911 capacitive touch driver for esp32-3248s035c implemented?

**Comment by htvekov** [comment]
Hi' @RossiRPI

The new GS-T3E 4" square display device also utilizes Goodix GT911 as touch driver.
So I guess the chance for that implementation is very high, when @fvanroie gets to it.

**Comment by RossiRPI** [comment]
There is a driver for it implemented in master but it tries to include Goodix.h which seems to not be part of the project so cant compile.

**Reply by fvanroie** [reply]
You need to add `${goodix.lib_deps}` to the `lib_deps` key (edited)
Check https://github.com/HASwitchPlate/openHASP/blob/master/user_setups/esp32/ttgo-lilygo-pi_ili9481.ini

Do you have the specs of the esp32-3248s035c? Is it for sure the GT911 or some other capacities touch panel?
Would it be possible to a scan of the I2C bus to check which ID the touch panel uses? Thanks.

**Comment by RossiRPI** [comment]
according to the LVGL example that is provided for the board it does use the GT911 and it uses 0xBA and 0xBB for the read and write addresses.

**Reply by RossiRPI** [reply]
How do I scan the I2C bus?

**Reply by fvanroie** [reply]
> How do I scan the I2C bus?

I'm using [this method from Adafruit](https://learn.adafruit.com/scanning-i2c-addresses/arduino).
```
[15:00:36.207][77812/109036 28][45832/45856  1] DRVR: Scanning I2C...
[15:00:36.225][77812/109036 28][45832/45856  1] DRVR:     *  Found device 0x5d
[15:00:36.239][77812/109036 28][45832/45856  1] DRVR: Scan complete
```
The ID `5D` corresponds with 0xBA and 0xBB  I²C slave device addresses.

**Comment by RossiRPI** [comment]
So... an I2C scan is actually part of the touch driver in OpenHASP and is done on initialization of the GT911 driver. you just have to build the driver with the log level set to verbose. so my build does seem to be picking up the device with the same address. 
[       2.468][54260/84248 35][45832/45856  1] DRVR: Scanning I2C...
[       2.485][54260/84248 35][45832/45856  1] DRVR:     *  Found device 0x5d
[       2.499][54260/84248 35][45832/45856  1] DRVR: Scan complete
Not sure why its not registering the touches though.

**Reply by RossiRPI** [reply]
Problem solved, there's a series 0 ohm resistor/jumper which is not fitted on any of my esp32-3248s035c boards. This resistor/jumper connects the GT911 interrupt line to the ESP32 hence why it wasn't working.
![image](https://user-images.githubusercontent.com/60947109/191951879-dccc768e-9f70-40cb-99af-6c737adaba2e.png)

**Reply by fvanroie** [reply]
Thanks! It's very weird that they would not populate this resistor for `TP_INT`, maybe `IO21` is shared with some other function/header...

Would you mind sharing your configuration and maybe a picture of the resistor/jumper needed on the PCB?

**Reply by RossiRPI** [reply]
Yea, IO21 which they use for the GT911's INT line also goes out on two connectors. Interestingly there are no jumpers to disconnect it from going to the connectors but there is (R25) which disconnects it from the GT911. I could have a look at the driver which they use in the demo but its safe to assume that they are running it in polled mode. 

Here's an image of the board with R25 and teh two connectors which can not be used once we connect IO25 to the GT911. 

![image](https://user-images.githubusercontent.com/60947109/192077030-39126836-ee61-403e-af9d-23bb431b6068.png)

Config is as follows:
```ini
[env:esp32-3248s035c_4MB]
extends = esp32_4mb_v2
board = esp32dev

build_flags =
    ${env.build_flags}
    ${esp32.build_flags}
    ${esp32.no_ps_ram}

;region -- TFT_eSPI build options ------------------------
    ${esp32.hspi}        ; Use HSPI hardware SPI bus
    -D USER_SETUP_LOADED=1
    ;-D LGFX_USE_V1=1
    -D ST7796_DRIVER=1
    -D TFT_ROTATION=0 ; 0=0, 1=90, 2=180 or 3=270 degree
    -D TFT_WIDTH=320
    -D TFT_HEIGHT=480
    -D TFT_CS=15      ;// Chip select control pin
    -D TFT_DC=2       ;// Data Command control pin
    -D TFT_RST=-1     ;// Reset pin (could connect to RST pin)
    -D TFT_BCKL=27    ;None, configurable via web UI (e.g. 2 for D4)
    -D SUPPORT_TRANSACTIONS    
    -D TOUCH_SDA=33
    -D TOUCH_SCL=32
    -D TOUCH_IRQ=21
    -D TOUCH_RST=25
    -D I2C_TOUCH_FREQUENCY=400000
    -D I2C_TOUCH_PORT=1
    -D I2C_TOUCH_ADDRESS=0x5D ; or 0x14
    -D TOUCH_DRIVER=0x0911 ; GT911 Capacitive I2C touch panel driver
    -D SPI_FREQUENCY=65000000
    -D SPI_TOUCH_FREQUENCY=2500000
    -D SPI_READ_FREQUENCY=20000000
;endregion

; -- Debugging options -----------------------------
;   -D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_DEBUG

;region -- Library options -------------------------------
lib_deps =
    ${env.lib_deps}
    ${esp32.lib_deps}
    ${arduino_esp32_v2.lib_deps}
    ${tft_espi.lib_deps}
    ;${lovyangfx.lib_deps}
    ${goodix.lib_deps}
    
lib_ignore =
    ${env.lib_ignore}
    ${esp32.lib_ignore}
    ${arduino_esp32_v2.lib_ignore}
;endregion
```

**Reply by eagl1** [reply]
My display is working on the demo code from the manufacturer and the touch is working. And the this R25 is open in my display board. Do you mean that the touch in your display didn't work out of the box ?

**Comment by mattiz** [comment]
Hi, I have a ESP32-3248S035 with a capacitive touch screen like the one on the picture above. I have tried the config above, and also esp32-3248s035c_4MB and esp32-3248s035c_4MB_lovyan, but I'm not able to get the touch screen to work. When I jump R25 the software get stuck in an reboot loop.

Any ideas of what could be wrong?

**Reply by PhantomSage** [reply]
@RossiRPI Just to get it right, before heating the soldering iron.
Is my interpretation right that you have to fit/jump R25 to get it working with OpenHASP?
Screen is working, but no calibration menu nor working capacitive touch right now :(

**Reply by mfitz2021** [reply]
Is it worth pointing out that touch works fine for me when I use the supplied demo but I can't get it to work with HASP.  It seems that it can detect touch because when I configure it to show the pointer it shows the pointer until I touch the display and then the pointer disappears.

**Reply by fvanroie** [reply]
Adding support for the [Sunton 2.8" with resistive touch](https://github.com/HASwitchPlate/openHASP/discussions/396#discussioncomment-4148236), had broken the capacitive touch for this board...
This should be fixed again in the latest build. No need to solder `GPIO21` anymore.

**Reply by exotsk** [reply]
Hello. Sorry, I didn’t understand whether this board works with a capacitive touch or does not work. Can I buy it for OpenHASP?

**Reply by fvanroie** [reply]
Both should work, with Capacitive being preferred over Resistive.

**Comment by eagl1** [comment]
Hi guys,

1. Here is my code with library files for this display, 4.3" and 7".
2. Install ESP32 2.0.17 board package.
3. Run my modified demo code. Just comment/uncomment the display model.
4. Put TAMC_GT911 & Arduino_GFX_TFT_eSPI in libraies folder.
5. Run the demo code.

[files.zip](https://github.com/user-attachments/files/22646624/files.zip)
